lib/rubyXL/cell.rb

```diff
@@ -104,7 +106,7 @@ def change_contents(data, formula_expression = nil)
         self.formula = RubyXL::Formula.new(:expression => formula_expression)
       else
         self.datatype = case data
-                        when Date, Integer, Float then nil
+                        when Date, Integer, Float, BigDecimal then nil
```

Comment:
I believe ```BigDecimal``` is inherited from ```Numeric```, so this should simply be ``when Date, Numeric then nil``